bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/container/ModuleResolver.java

```diff
@@ -470,6 +477,8 @@ class ResolveProcess extends ResolveContext implements Comparator<Capability>, E
 
 		class ResolveLogger extends Logger {
 			private Map<Resource, ResolutionException> errors = null;
+			public int totalPerm;
+			public int processedPerm;
```

Comment:
These could be stored in `org.eclipse.osgi.container.ModuleResolutionReport.Builder` and incremented in there so that all the data is stored there for the report.